### PR TITLE
MMT-2813 - Adding the Lambda project to generate TEA Configuration files from CMR metadata

### PR DESCRIPTION
**Pull request by jceaser**
Creating an AWS lambda project using serverless and written in python to generate TEA configuration files from CMR metadata.

This pull request will add the following endpoints to APIGateway

- configuration/tea
- configuration/capabilities
- configuration/debug - will be removed after some debugging is done
- configuration/tea/status
- configuration/tea/provider/{id}

Most interfaces will return JSON tea/provider will return a YAML file. Currently POCLOUD is the only provider with data and access will need to be granted, see Rosy Cordova to get access.

Deployment is not current active as this code is needed first.

### CODE REVIEW

File: tea-config-generator/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM node:latest
```
Comment (jaybarra) [inline tea-config-generator/Dockerfile:1]: Do we want to consider fixing this to a major version?

File: tea-config-generator/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Usage
+
+The [run.sh](run.sh) command will execute many of the stages of the software needed
+get the application started. To see the options supported, run `./run.sh -h`. To start the application, run in one terminal window: `./run.sh -o`. This will install the lambda functions and start the offline server. In another terminal window try
+the following commands:
+
+    curl 'http://localhost:3000/dev/configuration/tea/capabilities'
+    curl -I 'http://localhost:3000/dev/configuration/tea/status'
+    curl -H 'Cmr-Token: uat-token-value' 'http://localhost:3000/dev/configuration/tea/provider/POCLOUD'
```
Comment (jaybarra) [inline tea-config-generator/README.md:17]: Is this just an EDL bearer token? Can this header be `Authorization` instead?

File: tea-config-generator/run.sh [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      curl -H "Cmr-Token: ${token}" "${baseEndPoint}/configuration/tea/provider/POCLOUD"
+      echo
+      cprintf $GREEN '----'
+
+      echo
+      cprintf $GREEN 'Calling /tea/provider/POCLOUD again with just headers, may take 7 seconds'
+      curl -I -H "Cmr-Token: ${token}" "${baseEndPoint}/configuration/tea/provider/POCLOUD"
+      echo
+
+      #curl -s http://localhost:7000/local-bucket/index.html
```
Comment (jaybarra) [inline tea-config-generator/run.sh:162]: Comments should be removed here

**Review by jaybarra — CHANGES_REQUESTED** [review]

File: tea-config-generator/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[*]
```
Comment (daniel-zamora) [inline tea-config-generator/.editorconfig:1]: Do we want to merge in editorconfigs  here or just gitignore them?

File: tea-config-generator/README.md (hunk 2 above)
Comment (jceaser) [reply tea-config-generator/README.md:17]: good catch, I renamed that and missed the readme

File: tea-config-generator/run.sh (hunk 3 above)
Comment (jceaser) [reply tea-config-generator/run.sh:163]: I removed most, left a comment for the ones that I want in the file for documentation

File: tea-config-generator/.editorconfig (hunk 4 above)
Comment (jceaser) [reply tea-config-generator/.editorconfig:1]: I use editorconfig, so I would like it included.

File: tea-config-generator/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+6. Run some example URLS: `./run.sh -e`
+
+### Docker
+
+The Dockerfile defines a node image which includes python3 and serverless for
+use as a build and deployment environment when running under a CI/CD system.
+
+## Deploying to AWS
+
+1. Create a role with a title like `tea-config-<env>-<location>-lambda-role`
```
Comment (ygliuvt) [inline tea-config-generator/README.md:46]: Update this section now that you are granting the IAM role in serverless.yml.

**Review by ygliuvt — APPROVED** [review]
I only reviewed the deployment related code because the processing code is existing code and just being moved to CMR.

File: tea-config-generator/README.md (hunk 5 above)
Comment (jceaser) [reply tea-config-generator/README.md:46]: your right, will do that

**Review by jaybarra — APPROVED** [review]

**Pull request merged**